src/parallelization/hpxsimulator.cpp

```diff
@@ -6,12 +6,16 @@
 namespace LibGeoDecomp {
 namespace HpxSimulatorHelpers {
 
+hpx::lcos::local::promise<void> initPromise;
```

Comment:
On 02/12/2016 02:05 PM, Andreas Schäfer wrote:
> In src/parallelization/hpxsimulator.cpp
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3#discussion_r52738454>:
> 
>> @@ -6,12 +6,16 @@
>>  namespace LibGeoDecomp {
>>  namespace HpxSimulatorHelpers {
>>  
>> +hpx::lcos::local::promise<void> initPromise;
> 
>  1. Who is responsible for resetting this promise if the user wants to
>     create another HpxSimulator?
>  2. If the user is instantiating two HpxSimulators simultaneously this
>     single static variable creates a race condition.

ahhh ... is that the reason why you have a map for all the other promises?


> 
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/STEllAR-GROUP/libgeodecomp/pull/3/files#r52738454>.
> 


-- 
Thomas Heller
Friedrich-Alexander-Universität Erlangen-Nürnberg
Department Informatik - Lehrstuhl Rechnerarchitektur
Martensstr. 3
91058 Erlangen
Tel.: 09131/85-27018
Fax:  09131/85-27912
Email: thomas.heller@cs.fau.de